scripts/deploymentextensions/enablemobileapi/install.sh

```diff
@@ -0,0 +1,394 @@
+
+# cloud being deployed
+cloud="bvt"
+
+#############################################################################
+# parse the command line arguments
+
+parse_args() 
+{
+    while [[ "$#" -gt 0 ]]
+    do
+        arg_value="${2}"
+        shift_once=0
+
+        if [[ "${arg_value}" =~ "--" ]]; 
+        then
+            arg_value=""
+            shift_once=1
+        fi
+
+         # Log input parameters to facilitate troubleshooting
+        log "Option '${1}' set with value '"${arg_value}"'"
+
+        case "$1" in
+          --oxatools-public-github-accountname)
+            oxa_tools_public_github_account="${arg_value}"
+            ;;
+          --oxatools-public-github-projectname)
+            oxa_tools_public_github_projectname="${arg_value}"
+            ;;
+          --oxatools-public-github-projectbranch)
+            oxa_tools_public_github_projectbranch="${arg_value}"
+            ;;
+          --oxatools-public-github-branchtag)
+            oxa_tools_public_github_branchtag="${arg_value}"
+            ;;
+          --oxatools-repository-path)
+            oxa_tools_repository_path="${arg_value}"
+            ;;
+          --cluster-admin-email)
+            cluster_admin_email="${arg_value}"
+            ;;
+          --target-user)
+            target_user="${arg_value}"
+            ;;
+          --aad-webclient-id)
+            aad_webclient_id="${arg_value}"
+            ;;
+          --aad-webclient-appkey)
+            aad_webclient_appkey="${arg_value}"
+            ;;
+          --aad-tenant-id)
+            aad_tenant_id="${arg_value}"
+            ;;
+          --azure-subscription-id)
+            azure_subscription_id="${arg_value}"
+            ;;
+          --azure-resource-group)
+            azure_resource_group="${arg_value}"
+            ;;
+          --wait-interval-seconds)
+            wait_interval_seconds="${arg_value}"
+            ;;
+          --vmss-deployment-id)
+            vmss_deployment_id="${arg_value}"
+            ;;
+          --cloud)
+            cloud="${arg_value}"
+            ;;
+          --remote)
+            remote_mode=1
+            ;;
+          --debug)
+            debug_mode=1
+            ;;
+        esac
+
+        shift # past argument or value
+
+        if [ $shift_once -eq 0 ]; 
+        then
+            shift # past argument or value
+        fi
+
+    done
+}
+
+execute_remote_command()
+{
+    remote_execution_server_target=$1
+    remote_execution_target_user=$2
+
+    # build the command for remote execution (basically: pass through all existing parameters)
+    repository_parameters="--oxatools-public-github-accountname ""${oxa_tools_public_github_account}"" --oxatools-public-github-projectname ""${oxa_tools_public_github_projectname}"" --oxatools-public-github-projectbranch ""${oxa_tools_public_github_projectbranch}"" --oxatools-public-github-branchtag ""${oxa_tools_public_github_branchtag}"" --oxatools-repository-path ""${oxa_tools_repository_path}"""
```

Comment:
is this an exact copy of the params that invoked this bash file? if so, why not just pass "$@" ? it might require some work and testing, but i think it'll be less code and far less brittle